fala/common/results_view.py

```diff
@@ -29,20 +31,41 @@ def get(self, request, *args, **kwargs):
                 self.state = OtherJurisdictionState(region, form.cleaned_data["postcode"])
         else:
             if self.tailored_results:
-                # using CategoryMixin to access category_display_name & category_message, so we show this on SingleSearchErrorState view
+                # using CategoryMixin to access category_display_name & category_message, so we show this on CategorySearchErrorState view
                 self.setup_category(request, *args, **kwargs)
-                category_display_name = self.category_slug.replace("-", " ").title()
+                category_display_name = (self.category_slug or "").replace("-", " ").title()
                 category_message = CATEGORY_MESSAGES.get(self.category_slug, "")
 
-                # this is so we can use category_code & search_url, when conducting a search from SingleSearchErrorState view
-                category_code = self.request.GET.get("categories", "")
-                search_url = reverse("results")
+                # this is so we can use category_code, sub_category_code & search_url, when conducting a search from CategorySearchErrorState view
+                category_code = (
+                    self.request.GET.getlist("categories")[0]
```

Comment:
```
categories = self.request.GET.getlist("categories", [])
category = categories[0] if categories else ""
```

Removes unnecessary len(...) > 0 check.

Can you look through your code wherever you do > 0 and tweak where possible.

Reply:
Yep, now updated. Good suggestion.